common/ayon_common/utils.py

```diff
@@ -88,3 +100,422 @@ def get_ayon_launch_args(*args):
+
+    Returns:
+        list[dict[str, Any]]: Executable info matching version.
+    """
+
+    executables_info = get_executables_info()
+    return [
+        item
+        for item in executables_info.get("available_versions", [])
+        if item.get("version") == version
+    ]
+
+
+def get_executable_paths_by_version(version, only_available=True):
+    """Get executable paths by version.
+
+    Returns:
+        list[str]: Paths to executables.
+    """
+
+    output = []
+    for item in get_executables_info_by_version(version):
+        executable = item.get("executable")
+        if not executable:
+            continue
+
+        # Skip if executable was not found
+        if only_available and not os.path.exists(executable):
+            continue
+
+        output.append(executable)
+    return output
+
+
+def cleanup_executables_info():
+    """Remove executables that do not exist anymore."""
+
+    new_executables_info = []
+    changed = False
+    for item in get_executables_info():
+        executable = item.get("executable")
+        if not executable or not os.path.exists(executable):
+            changed = True
+            continue
+
+        version = load_executable_version(executable)
+        if version and item.get("version") != version:
+            changed = True
+            item["version"] = version
+        new_executables_info.append(item)
+
+    if changed:
+        store_executables_info(new_executables_info)
+
+
+class _Cache:
+    downloads_dir = 0
+
+
+def _get_linux_downloads_dir():
+    return subprocess.run(
+        ["xdg-user-dir", "DOWNLOAD"],
+        capture_output=True, text=True
+    ).stdout.strip("\n")
+
+
+def _get_windows_downloads_dir():
+    import ctypes
+    from ctypes import windll, wintypes
+
+    class GUID(ctypes.Structure):  # [1]
+        _fields_ = [
+            ("Data1", wintypes.DWORD),
+            ("Data2", wintypes.WORD),
+            ("Data3", wintypes.WORD),
+            ("Data4", wintypes.BYTE * 8)
+        ]
+
+        def __init__(self, uuid_):
+            ctypes.Structure.__init__(self)
+            (
+                self.Data1,
+                self.Data2,
+                self.Data3,
+                self.Data4[0],
+                self.Data4[1],
+                rest
+            ) = uuid_.fields
+            for i in range(2, 8):
+                self.Data4[i] = rest >> (8 - i - 1) * 8 & 0xff
+
+    pathptr = ctypes.c_wchar_p()
+    guid = GUID(WIN_DOWNLOAD_FOLDER_ID)
+    if windll.shell32.SHGetKnownFolderPath(
+        ctypes.byref(guid), 0, 0, ctypes.byref(pathptr)
+    ):
+        return None
+    return pathptr.value
+
+
+def _get_macos_downloads_dir():
+    """Get downloads directory on MacOS.
+
+    Notes:
+        By community forum '~/Downloads' is right way, which is default.
+
+    Returns:
+        Union[str, None]: Path to downloads directory or None if not found.
+    """
+
+    return None
+
+
+def get_downloads_dir():
```

Comment:
Why don't we just make a temp directory and use it as "Downloads" ?

Reply:
If installation fails, you don't have to download it again (if it's in Downloads) and do a manuall installation.